src/app.provider.ts

```diff
@@ -66,13 +71,17 @@ export abstract class AppProvider {
     app: INestApplication,
   ) => void)[];
 
-  public async provide(module: any): Promise<INestApplication> {
+  public async provide(
+    module: DynamicModule | TestingModule,
```

Comment:
I would not restrict the module to be a `DynamicModule` here.

Given that NestJS expected `any` in `NestFactory.create` we can have a generic type instead.


```typescript
export abstract class AppProvider<T> {
  ...
  public async provide(module: T): Promise<INestApplication> {
    ...
  }

  protected abstract getApp(module: T): Promise<INestApplication>;
}
```

Reply:
Good idea! I've adjusted it to use generics in 646f900c7186f2e596033fe23ebb9e1a8202c7bf.